src/pharmpy/modeling/allometry.py

```diff
@@ -84,6 +87,15 @@ def add_allometry(
     S₁ = V
 
     """
+    if allometric_variable is None:
+        for col in model.datainfo:
+            if col.descriptor == 'body weight':
+                allometric_variable = col.name
```

Comment:
There is a nice `else' for loops that could be used here.

```python
for col in model.datainfo:
    if col.descriptor == 'body weight':
        allometric_variable = col.name
        break
else:
    raise ValueError(...)
```

Even better would be to use the `descriptorix` of `DataInfo` instead of looping

Reply:
Thanks for the comment! I redid the change using `descriptorix` as per your suggestion. Let me know if there are any other comments :)